warehouse/core/src/main/java/datawave/ingest/mapreduce/job/SplittableRFileInputFormat.java

```diff
@@ -0,0 +1,137 @@
+package datawave.ingest.mapreduce.job;
+
+import static org.apache.accumulo.core.conf.Property.TABLE_CRYPTO_PREFIX;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.accumulo.core.crypto.CryptoFactoryLoader;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.file.FileSKVIterator;
+import org.apache.accumulo.core.file.blockfile.impl.CachableBlockFile;
+import org.apache.accumulo.core.file.rfile.RFile;
+import org.apache.accumulo.core.spi.crypto.CryptoEnvironment;
+import org.apache.accumulo.core.spi.crypto.CryptoService;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.apache.log4j.Logger;
+
+public class SplittableRFileInputFormat extends RFileInputFormat {
+    private static final Logger log = Logger.getLogger(SplittableRFileInputFormat.class);
+    public static final String MIN_BLOCKS_PER_SPLIT = SplittableRFileInputFormat.class.getName() + ".minBlocksPerSplit";
+
+    @Override
+    protected boolean isSplitable(JobContext context, Path filename) {
+        return true;
+    }
+
+    @Override
+    public RecordReader<Key,Value> createRecordReader(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+        return new SplittableRFileRecordReader();
+    }
+
+    /**
+     * Create rfile splits for the configured job. See <code>FileInputFormat</code>
+     *
+     * @param job
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public List<InputSplit> getSplits(JobContext job) throws IOException {
+        Configuration config = job.getConfiguration();
+
+        List<InputSplit> rfileSplits = new ArrayList<>();
+        log.info("getting splits for job");
+        // get the configured directories/files
+        for (InputSplit inputSplit : super.getSplits(job)) {
+            if (!(inputSplit instanceof FileSplit)) {
+                throw new IllegalArgumentException("Must have file splits");
+            }
+
+            // for each file get the index blocks
+            FileSplit fileSplit = (FileSplit) inputSplit;
+            log.info("creating splits for: " + fileSplit);
```

Comment:
This logline dupes the one in `getSplits(config, fileSplit)`

Reply:
will remove